mkfs/rootdir.c

```diff
@@ -403,10 +463,10 @@ static int add_file_items(struct btrfs_trans_handle *trans,
 			goto end;
 		}
 
-		ret_read = pread(fd, buffer, st->st_size, bytes_read);
+		ret_read = pread(fd, buffer, st->st_size, 0);
 		if (ret_read == -1) {
-			error("cannot read %s at offset %llu length %llu: %m",
-				path_name, bytes_read, (unsigned long long)st->st_size);
+			error("cannot read %s at offset %u length %llu: %m",
+				path_name, 0, (unsigned long long)st->st_size);
```

Comment:
Since you are here, you can also fix the length output, without using a forced `(unsigned long long)` conversion, but `%zu`.